packages/bundler-plugin-core/src/utils/getPreSignedURL.ts

```diff
@@ -55,8 +55,30 @@ export const getPreSignedURL = async ({
 
       requestBody.git_service = foundGitService;
     }
-  } else if (uploadToken) {
-    headers.set("Authorization", `token ${uploadToken}`);
+  } else if (output.oidc?.useGitHubOIDC && Core) {
+    if (serviceParams?.service !== "github-actions") {
+      red("OIDC is only supported for GitHub Actions");
+      throw new BadOIDCServiceError(
+        "OIDC is only supported for GitHub Actions",
+      );
+    }
+
+    url = output.oidc.OIDCEndpoint;
```

Comment:
We should remove this line `url = output.oidc.OIDCEndpoint;`

The fact that the `audience` and the `apiUrl` are the same value of `https://api.codecov.io` is a coincidence and we should track 2 different variables for them. Our implementation of the JWT validation in codecov-api happens to expect that same value, but we shouldn't couple that current behavior to the code here.

Within this function (`getPreSignedURL`) we only ever want to actually hit the `apiUrl` which is the server that exposes the bundle analysis upload endpoint (`https://api.codecov.io/upload/bundle_analysis/v1`).